### PR TITLE
Add guest blog post on integrating ZAP with Noir for enhanced DAST

### PR DESCRIPTION
**Pull request by hahwul**
Hi @psiinon ,

This is the PR for the guest ZAP blog post we talked a few days ago. The post covers how to use Noir when working with ZAP. Since I'm not a native English speaker, there might be some awkward phrasing, so please take a quick look and let me know if anything needs to be corrected.

And.. I've written the post's date according to my local timezone (Seoul/GMT+9), since there wasn’t any specific timezone information in the Hugo config. If there's an official timezone you prefer, we might need to adjust it.

Hope you have a great day!

### CODE REVIEW

**Comment by psiinon** [comment]
Wow - great blog post!
Only comment - can you change the date to today? 😁 

<details><summary>Screenshot (to help other reviewers)</summary>
<p>

Screenshot 2024-11-11 at 11-09-32 ZAP – Powering Up DAST with ZAP and Noir

</p>
</details>

File: site/content/blog/2024-11-10-powering-up-dast-with-zap-and-noir/index.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+date: "2024-11-10"
+summary: "Integrating Noir, a tool for discovering hidden endpoints in source code, with ZAP enhances dynamic application security testing (DAST)."
+type: post
+tags:
+  - blog
+  - automation
+  - recon
+authors:
+  - hahwul
+---
```
Comment (kingthorin) [inline site/content/blog/2024-11-10-powering-up-dast-with-zap-and-noir/index.md:12]: I'd suggest setting the title image for the social preview.

File: site/content/blog/2024-11-10-powering-up-dast-with-zap-and-noir/index.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+ZAP’s scanning capabilities make it ideal for DAST implementations in DevSecOps, and integrating Noir with ZAP enhances this process. Noir can export results in multiple formats, including JSON and YAML, with support for OAS specifications. An example of generating OAS output with Noir is shown below:
+
+```sh
+noir -b <SOURCE> -f oas3 --no-log -o <OUTPUT>
+```
+
+In this article, we will use the OAS format.
+
+#### Command line
```
Comment (kingthorin) [inline site/content/blog/2024-11-10-powering-up-dast-with-zap-and-noir/index.md:59]:
```suggestion
#### Command Line
```

File: site/content/blog/2024-11-10-powering-up-dast-with-zap-and-noir/index.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+# Automation scan with endpoints
+./zap.sh -openapifile ./doc.json \
+    -openapitargeturl <TARGET> \
+    -cmd -autorun zap.yaml <any other ZAP options>
+```
+
+![](./images/output.png)
+
+#### Docker image
```
Comment (kingthorin) [inline site/content/blog/2024-11-10-powering-up-dast-with-zap-and-noir/index.md:75]:
```suggestion
#### Docker Image
```

File: site/content/blog/2024-11-10-powering-up-dast-with-zap-and-noir/index.md (hunk 1 above)
Comment (kingthorin) [reply site/content/blog/2024-11-10-powering-up-dast-with-zap-and-noir/index.md:12]: The syntax is:
```
images:
- https://www.zaproxy.org/blog/2023-08-25-community-tips-and-tricks/images/zapbot.TandT.gif
addSocialPreview: true
```

Comment (thc202) [reply site/content/blog/2024-11-10-powering-up-dast-with-zap-and-noir/index.md:12]: The `addSocialPreview` is true by default for blog posts.

**Review by kingthorin — APPROVED** [review]

**Comment by kingthorin** [comment]
If the DCO is going to be a headache you can just let us know that you do agree to: https://developercertificate.org/

**Comment by hahwul** [comment]
@kingthorin I agree.

**Review by thc202 — APPROVED** [review]

**Comment by kingthorin** [comment]
Thanks 

image

**Comment by thc202** [comment]
(First commit had DCO.)

**Comment by thc202** [comment]
Thank you!

**Pull request merged**